// Copied from hs-proto's PegasusShared protos


enum RuleType {
	RULE_NONE = 0;
	RULE_CHOOSE_HERO = 1;
	RULE_CHOOSE_DECK = 2;
}

message GameSetupRule {
	required int32 id = 1;
	required RuleType rule_type = 2;
	required int64 data1 = 3;
	required int64 data2 = 4;
	required int64 data3 = 5;
}
message LocalizedStringValue {
	required int32 locale = 1;
	required string value = 2;
}

message LocalizedString {
	required string key = 1;
	optional string deprecated_value = 2;
	optional int32 deprecated_locale = 3;
	repeated LocalizedStringValue values = 4;
}

message Vector2 {
	required float x = 1 [default = 0];
	required float y = 2 [default = 0];
}

message ScenarioDbRecord {
	required int32 id = 1;
	optional string note_desc = 2;
	required int32 num_players = 3;
	required int64 player1_hero_card_id = 4;
	required int64 player2_hero_card_id = 5;
	required bool is_expert = 6;
	required int32 adventure_id = 7;
	optional int32 adventure_mode_id = 8;
	required int32 wing_id = 9;
	required int32 sort_order = 10;
	optional int64 client_player2_hero_card_id = 11;
	optional string tavern_brawl_texture = 12;
	optional string tavern_brawl_texture_phone = 13;
	optional Vector2 tavern_brawl_texture_phone_offset = 14;
	optional bool is_coop = 15;
	optional int32 deck_ruleset_id = 16;
	optional int32 rule_type = 17;
	repeated LocalizedString strings = 100;
	repeated GameSetupRule rules = 101;
}
